vortex-array/src/arrays/listview/reset.rs

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use vortex_dtype::{Nullability, match_each_integer_ptype};
+use vortex_error::{VortexExpect, VortexResult};
+
+use crate::arrays::ListViewArray;
+use crate::builders::{ArrayBuilder, PrimitiveBuilder, builder_with_capacity};
+use crate::{ArrayRef, Canonical, IntoArray, OffsetPType, compute};
+
+impl ListViewArray {
+    /// Create a new version of this array by adjusting `offsets` to start at 0 and removing any
+    /// `elements` not referenced by the `offsets` + `sizes`.
+    ///
+    /// Unlike [`ListArray`], the offsets of a `ListViewArray` are allowed to be out-of-order. This
+    /// means that we cannot simply subtract the first offset from the rest of the offsets (because
+    /// we don't know where the used elements actually are, and we may reorder some scalars).
+    ///
+    /// [`ListArray`]: crate::arrays::ListArray
+    #[allow(clippy::cognitive_complexity)]
+    pub fn reset_offsets(&self) -> VortexResult<Self> {
```

Comment:
I think `reset_offsets` made sense for ListArray since there the change is to have offsets start at 0. In this case you are removing values that are not reference by any offset+size pair. I would call this `compact_values`